File: 백준/Silver/1874. 스택 수열/스택 수열.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());

        int [] arr = new int[n];
        for (int i = 0; i < n; i++){
            arr[i] = Integer.parseInt(br.readLine());
        }
        Stack<Integer> stack = new Stack<>();
        StringBuffer sb = new StringBuffer();
        int num  = 1;
        boolean result = true;
        for (int su : arr) {
            if (su >= num) {
                while (su >= num) {
                    stack.push(num++);
                    sb.append("+\n");
                }
                stack.pop();
                sb.append("-\n");
            } else {
                int N = stack.pop();
                if (N > su) {
                    System.out.println("NO");
                    result = false;
                    break;
                } else {
                    sb.append("-\n");
                }
            }
        }
     if (result) System.out.println(sb.toString());
    }
}

File: 백준/Silver/1874. 스택 수열/스택 수열.py
import sys
input = sys.stdin.readline
n = int(input())
stack = []
result = []
flag = 0
now = 1
for i in range(n):
    num = int(input())
    while now <= num:
        stack.append(now)
        result.append("+")
        now += 1
    if stack[-1] == num:
        stack.pop()
        result.append("-")
    else:
        print("NO")
        flag = 1
        break
if flag == 0:
    for i in result:
        print(i)